pkg/controller/jobs/statefulset/statefulset_webhook_test.go

```diff
@@ -155,7 +155,9 @@ func TestDefault(t *testing.T) {
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
-			features.SetFeatureGateDuringTest(t, features.LocalQueueDefaulting, tc.localQueueDefaulting)
+			if !tc.localQueueDefaulting {
```

Comment:
I'd revert this change and keep the old change.

You can always set a feature gate to false/true in the tests.